util/perl/TLSProxy/Proxy.pm

```diff
@@ -93,13 +93,42 @@ sub new_dtls {
 
 sub init
 {
+    require IO::Socket::IP;
     my $class = shift;
     my ($filter,
         $execute,
         $cert,
         $debug,
         $isdtls) = @_;
 
+    my $test_client_port;
+
+    # Sometimes, our random selection of client ports gets unlucky
+    # And we randomly select a port thats already in use.  This causes
+    # this test to fail, so lets harden ourselves against that by doing
+    # a test bind to the randomly selected port, and only continue once we
+    # find a port thats available.
+    while (1) {
+        my $test_client_addr = $have_IPv6 ? "[::1]" : "127.0.0.1";
```

Comment:
This line can be moved out of the `while` block, and the result reused below for `proxy_addr`